ocrd_anybaseocr/cli/ocrd_anybaseocr_cropping.py

```diff
@@ -44,408 +44,410 @@
+
+
+    def crop_area(self, textarea, binImg, rgb):
+        height, width, _ = binImg.shape
+
+        textarea = np.unique(textarea, axis=0)
+        i = 0
+        tmp = []
+        areas = []
+        while i < len(textarea):
+            textarea = [list(x) for x in textarea if x not in tmp]
+            tmp = []
+            if len(textarea) == 0:
+                break
+            maxBox = textarea[0]
+            for chkBox in textarea:
+                if maxBox != chkBox:
+                    x11, y11, x12, y12 = maxBox
+                    x21, y21, x22, y22 = chkBox
+                    if ((x11 <= x21 <= x12) or (x21 <= x11 <= x22)):
+                        tmp.append(maxBox)
+                        tmp.append(chkBox)
+                        maxBox = [min(x11, x21), min(y11, y21),
+                                  max(x12, x22), max(y12, y22)]
+            if len(tmp) == 0:
+                tmp.append(maxBox)
+            x1, y1, x2, y2 = maxBox
+            areas.append(maxBox)
+            cv2.rectangle(rgb, (x1, y1), (x2, y2), (255, 0, 0), 2)
+            i = i+1
+
+        textarea = np.unique(areas, axis=0).tolist()
+        if len(textarea) > 0:
+            textarea = self.filter_area(textarea, binImg)
+        if len(textarea) > 1:
+            textarea = self.marge_columns(textarea)
+            # print textarea
+
+        if len(textarea) > 0:
+            textarea = sorted(textarea, key=lambda x: (
+                (x[2]-x[0])*(x[3]-x[1])), reverse=True)
+            # print textarea
+            x1, y1, x2, y2 = textarea[0]
+            x1 = x1-20 if x1 > 20 else 0
+            x2 = x2+20 if x2 < width-20 else width
+            y1 = y1-40 if y1 > 40 else 0
+            y2 = y2+40 if y2 < height-40 else height
+
+            #self.save_pf(base, [x1, y1, x2, y2])
+
+        return textarea
+
+    def process(self):
+        for (n, input_file) in enumerate(self.input_files):            
+            pcgts = page_from_file(self.workspace.download_file(input_file))                
+            fname = pcgts.get_Page().imageFilename
+            img = self.workspace.resolve_image_as_pil(fname)
+            #fname = str(fname)
+            print("Process file: ", fname)
+            base, _ = ocrolib.allsplitext(fname)
+
+            img_array = ocrolib.pil2array(img)
+            img_array_bin = np.array(img_array > ocrolib.midrange(img_array), 'i')
+
+            lineDetectH = []
+            lineDetectV = []
+            img_array_rr = self.remove_rular(img_array)
+            textarea, img_array_rr_ta, height, width = self.detect_textarea(img_array_rr)
+            self.parameter['colSeparator'] = int(width * self.parameter['colSeparator'])
+
+            if len(textarea) > 1:
+                textarea = self.crop_area(textarea, img_array_bin, img_array_rr_ta)
+                if len(textarea) == 0:
+                    min_x ,min_y, max_x, max_y = self.select_borderLine(img_array_rr, lineDetectH, lineDetectV)
```

Comment:
I must say, I do not understand enough of the original code. It looks awful all over, and it seems to use Numpy arrays for passing objects by reference. But in this particular place, no output of `select_borderLine` got used anywhere. So I guess you must have reconstructed the originally intended meaning here.